developing/on-call/README.md

```diff
@@ -0,0 +1,193 @@
+  to cover for you while you are away.
+* Despite the expectation you will be the first responder as the person
+  on-call, *this does not mean you are expected to go it alone*. If you
+  get an alert, and you can't figure out what is going on within 15
+  minutes and you believe the impact is such that it needs to be addressed
+  immediately, you should feel free to page your secondary for assistance.
+  If you are still stuck (or you *were* the secondary), you should feel
+  free to call upon your lead or a known subject matter expert (SME).
+* If you are *not* on-call, you should refrain from responding to alerts
+  even if you see them in Slack or elsewhere. By doing so, you can reduce
+  your own interrupts. However, if you believe you might be responsible,
+  or know the on-call person is dealing with another higher-priority issue
+  and want to assist, *let the on-call engineer know* and then make sure
+  you take ownership of the alert in PagerDuty. Remember that they likely
+  already got the alert notification and make sure they have acknowledged
+  that you will be taking care of the alert before taking action, so that
+  you are not working at cross-purposes or duplicating work.
+* You should make sure that you are keeping a persistent record of alerts
+  and/or incidents each day. This can be as simple as a Google Doc filled
+  out at the end of the day, but it should record at least the time of the
+  alert, the alert that fired, and what was done to address the alert (even
+  if that is "the alert went away on its own"). This serves as a way to
+  pass knowledge onto the other on-call engineers or the next shift, and
+  allows us to look at the previous week or month for alerts that are
+  particularly troublesome.
+
+## Escalation and Notification Policies
+
+In PagerDuty terms, an escalation policy determines how an alert will
+proceed if it is not acknowledged; a notification policy is something
+which is set for each engineer individually that determines how they will
+be notified if they receive an alert.
+
+* PagerDuty and similar products can be set up to send notifications to
+  Slack for each alert; we recommend doing so. Using the [PagerDuty
+  integration](https://www.pagerduty.com/docs/guides/slack-integration-guide/)
+  will also allow engineers to acknowledge or resolve alerts from Slack
+  if they so choose.
+* Engineers should have notification policies set to ensure that they
+  will be notified within the expect five minute response window. This
+  should use multiple notification methods to make sure things don't fall
+  through the cracks; at least one method should notify you immediately
+  that the alert has fired. Keep in mind that an acknowledgement will break
+  the notification chain. An example might be:
+  * Immediately after the alert, notify me by push notification and email.
+  * 1 minute later, notify via SMS (in case data coverage is bad).
+  * 5 minutes after the alert, notify via voice call.
```

Comment:
This feels tight. For most projects in my lifetime I've seen this:

* Immediately via apps and email
* 5 minutes SMS
* 10-15 minutes voice call

And then at 20 minutes have it escalate. Each project will obviously have different tolerances but I've been in a situation where I'm just about to check my app and my texts start blowing up only to start reading my texts and get called. If it turns out folks really aren't responding frequently within 10-15 minutes then we probably have a different problem on our hands.

Reply:
The problem with that, as I outline in the document, is that using that policy basically guarantees you will drop to three-nines of availability if you have a critical issue that falls through. Obviously, this all depends on the tolerances of your project -- I will admit my last on-call experience was for a 24/7 web service where we were aiming for four-nines, so tolerances were tight. I would rather say the best practice is tight, and then have the engineering and product folks decide to loosen it, rather than the other way around.

Reply:
I'm going to say that if MilMove gets 90% uptime we'll be in the clear. Maybe a recommendation chart that looks like this would be nicer (numbers made up):

| Notifications | 99.99% | 99.9% | 99% |
| --- | --- | ---| --- |
| app/email | immediately | immediately | immediately |
| SMS | 1 | 5 | 10 |
| Voice | 5 | 10 | 15 |

Reply:
I guess I'm trying to say that we should align our on-call with the expectation of the customer based on the SLA instead of making all teams attempt to get two or more 9's.

Reply:
I think this was just an example, not a suggestion. It would depend on contract agreements Truss has with the client around response time, so examples should also.

Reply:
I think a chart with examples would be nice, so maybe these totally made up numbers but based on project:

| Notifications | MilMove | SABER | CMS |
| --- | --- | ---| --- |
| app/email | immediately | immediately | immediately |
| SMS | 5 | 5 | 10 |
| Voice | 10 | 15 | 10 |